core/node/consensus/src/storage/testonly.rs

```diff
@@ -189,3 +195,111 @@ impl ConnectionPool {
         Ok(())
     }
 }
+
+/// A struct for writing to consensus L2 contracts.
+#[derive(Debug)]
+pub struct VMWriter {
+    pool: ConnectionPool,
+    node: StateKeeper,
+    account: Account,
+}
+
+impl VMWriter {
+    /// Constructs a new `VMWriter` instance.
+    pub fn new(pool: ConnectionPool, node: StateKeeper, account: Account) -> Self {
+        Self {
+            pool,
+            node,
+            account,
+        }
+    }
+
+    /// Deploys the consensus registry contract and adds nodes to it.
+    pub async fn deploy_and_add_nodes(
+        &mut self,
+        ctx: &Ctx,
+        owner: Address,
+        nodes: &[&[Token]],
```

Comment:
can the conversion be done inside the function, passing &[&[Token]] is very loosely typed.